core/os/input.cpp

```diff
@@ -80,6 +80,7 @@ void Input::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("get_joy_axis_index_from_string", "axis"), &Input::get_joy_axis_index_from_string);
 	ClassDB::bind_method(D_METHOD("start_joy_vibration", "device", "weak_magnitude", "strong_magnitude", "duration"), &Input::start_joy_vibration, DEFVAL(0));
 	ClassDB::bind_method(D_METHOD("stop_joy_vibration", "device"), &Input::stop_joy_vibration);
+	ClassDB::bind_method(D_METHOD("vibrate", "duration_ms"), &Input::vibrate, DEFVAL(500));
```

Comment:
I wonder if Input is the right place for this, but given that there is already other device-related things like gravity/gyroscope, etc., I guess it's fine for the time being.

Reply:
Should the name be made more explicit, to avoid possible confusion with joypad vibration? I'm not sure what would be clear though, maybe "vibrate_device"? Could still be a bit confusing though as joypads are also devices, but I guess it's a bit more explicit nevertheless. Could be "vibrate_handheld_device" but that starts being quite verbose.

Suggestions welcome :)

Reply:
I had an exact same thought. but couldn't find explicit name for it.
I also welcome suggestions. :)

Reply:
Unity's API seems to have `Handheld.Vibrate`, maybe we can use `vibrate_handheld`? Joypads are also handheld devices of course, but I think it should be relatively clear to most devs that this would be about the host mobile/tablet device.

Reply:
I'm ok with it.

Reply:
Let's go with `vibrate_handheld` then, I'll merge once it's rebased.

Reply:
@akien-mga done :)